docs/guides/protect-page-login/flutter-web-redirect.mdx

```diff
@@ -0,0 +1,190 @@
+---
+id: flutter-web-redirect
+title: Protect a Page with Login in Flutter Web
+sidebar_label: Flutter Web application
+---
+
+```mdx-code-block
+import Teaser from './_common/teaser.mdx'
+import mp4 from './flutter_web_redirect/screencast.mp4'
+import webm from './flutter_web_redirect/screencast.webm'
+
+<Teaser
+  framework={<a href="https://flutter.dev/">Flutter</a>}
+  mp4={mp4}
+  webm={webm}
+/>
+```
+
+## Preface
+
+From experience, we at Ory have noticed many resources do not fully explain
+which authentication method is the correct method for that technology or use
+case. With frameworks that support so many devices, it becomes difficult to
+understand where to use, for example, session cookies, tokens, or even OAuth2
+flows.
+
+We won't cover such a wide range of topics in one guide and will thus split them
+up into different projects/guides.
+
+This guide will specifically focus on **Flutter Web using browser redirects to
+set a session cookie**. To do this, we will rely on the Ory Managed UI for
+login/registration instead of building our own.
+
+## Create Flutter Web App
+
+:::note
+
+This guide assumes you already have flutter installed on your system. Please
+follow the official installation guide if you have not at
+https://docs.flutter.dev/get-started/install
+
+:::
+
+The below command will create a basic Flutter example application. We will just
+add to it an authentication layer to keep the guide simple.
+
+```shell-session
+flutter create myapp
+cd myapp
+```
+
+For HTTP requests we will use the [dio](https://github.com/flutterchina/dio)
+package.
+
+All the packages we use throughout this guide are listed in our `pubspec.yaml`.
+
+```mdx-code-block title="pubspec.yaml"
+import pubspec from '!!raw-loader!../../../code-examples/protect-page-login/flutter_web_redirect/pubspec.yaml'
+import CodeBlock from '@theme/CodeBlock'
+
+<CodeBlock language="yaml">{pubspec}</CodeBlock>
+```
+
+Package sources:
+
+- https://github.com/flutterchina/dio
+- https://github.com/java-james/flutter_dotenv
+
+Next let us create an Authentication service in our `lib/services` directory,
+which we will use to query the Ory APIs for session information.
+
+```shell-session
+touch lib/services/auth.dart
+```
+
+```mdx-code-block title="lib/services/auth.dart"
+import auth from '!!raw-loader!../../../code-examples/protect-page-login/flutter_web_redirect/lib/services/auth.dart'
+
+<CodeBlock language="dart">{auth}</CodeBlock>
+```
+
+Then we need to create an `.env` file in the root of our project to hold the
+`ORY_BASE_URL` variable. This would be the Ory proxy URL (e.g.
+http://localhost:3005/.ory) and the path to the Ory API `/api/kratos/public`.
```

Comment:
`Then we need` -> no

Remember, you're writing instructions for somebody. 

`e.g.` -> don't use. Use `For example` instead.